### PR TITLE
fix: replace Decimal `setcontext` with `localcontext` context manager

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR replaces all instances of Decimal's `setcontext` with the `localcontext` context manager.

Tests were failing for some weird dependency reason, so this PR also updates some dev dependencies as well.

### Context of Change

It's better practice to use a context manager than to set global state. Plus, it's less likely to cause problems.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Test Plan

CI passes (tests don't pass if the context is set incorrectly).

### CODE REVIEW

**Review by mDuo13 — APPROVED** [review]
Not necessary as part of this PR, but might it be helpful to export the XRP and token contexts for clients to use for their own calculations as well?

**Review by JST5000 — APPROVED** [review]
LGTM!

File: xrpl/core/binarycodec/types/amount.py [hunk 1]
```diff
@@ -24,8 +24,9 @@
 _MAX_MANTISSA: Final[int] = 10 ** 16 - 1
 
 # Configure Decimal
-setcontext(
-    Context(prec=_MAX_IOU_PRECISION, Emax=_MAX_IOU_EXPONENT, Emin=_MIN_IOU_EXPONENT)
+# TODO: export this context so others can use it
+_AMOUNT_CONTEXT: Final[Context] = Context(
+    prec=_MAX_IOU_PRECISION, Emax=_MAX_IOU_EXPONENT, Emin=_MIN_IOU_EXPONENT
```
Comment (ledhed2222) [inline xrpl/core/binarycodec/types/amount.py:29]: move this to `constants.py` and get rid of the todo

**Review by ledhed2222 — APPROVED** [review]

**Review by khancode — APPROVED** [review]
LGTM

**Review by ledhed2222 — APPROVED** [review]

**Pull request merged**